### PR TITLE
allow overriding the default sdk http client settings and ability to configure retries

### PR DESCRIPTION
**Pull request by lavanyachennupati**
allow overriding the default sdk http client settings and ability to configure retries.

Provides ability to override the [default sdk http configuration](https://github.com/aws/aws-sdk-java-v2/blob/master/http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java#L137) and ability to configure [retries](https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/RetryPolicy.java)

### CODE REVIEW

**Review by dmuino — APPROVED** [review]

File: iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java [hunk 1]
```diff
@@ -296,19 +350,20 @@ private SdkAsyncHttpService createAsyncHttpService(String name) {
   public <T> T newInstance(String name, Class<T> cls, String accountId) {
     try {
       SdkHttpService service = createSyncHttpService(name);
+      SdkAsyncHttpService asyncService = createAsyncHttpService(name);
```
Comment (brharrington) [inline iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java:353]: Why was this moved up? It looks like it is only needed for the async case.

File: iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java [hunk 2]
```diff
@@ -88,6 +95,18 @@ private void setIfPresent(
     }
   }
 
+  private void setRetriesIfPresent(Config cfg, ClientOverrideConfiguration.Builder builder) {
+    if (cfg.hasPath("retries")) {
```
Comment (brharrington) [inline iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java:99]: How likely do you think it is that we might want to have a way to more generically specify other aspects of the retry policy in the future? Perhaps a config structure like:

```
retry-policy {
  num-retries = 5
}
```

That would make it easier to extend later if we wanted to have some of the other parts configurable.

File: iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java (hunk 1 above)
Comment (lavanyachennupati) [reply iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java:353]: No reason. I moved it outside when trying out some changes and forgot to clean it up when it's no longer need outside the async case. Will move it back inside the `AwsAsyncClientBuilder` case

File: iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java (hunk 2 above)
Comment (lavanyachennupati) [reply iep-module-aws2/src/main/java/com/netflix/iep/aws2/AwsClientFactory.java:99]: > How likely do you think it is that we might want to have a way to more generically specify other aspects of the retry policy in the future?
I can see cases that might be needed (though I expect it to be rare). I like the idea of a config structure so we can easily extend it, if needed. Thanks! Will push the change.

**Review by brharrington — APPROVED** [review]
